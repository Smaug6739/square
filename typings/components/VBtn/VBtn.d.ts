declare const props: Readonly<{
    onClick?: () => void;
    value?: string;
    enabled?: boolean;
    loading?: boolean | string;
    icon?: string;
}>;
import type { DefineComponent } from 'vue';
export declare type component = DefineComponent<typeof props, {}, any>;
export {};
