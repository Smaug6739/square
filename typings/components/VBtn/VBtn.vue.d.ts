declare const _default: {
    new (...args: any[]): {
        $: import("vue").ComponentInternalInstance;
        $data: {};
        $props: Partial<{}> & Omit<Readonly<import("vue").ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{
            onClick?: () => void;
            value?: string;
            enabled?: boolean;
            loading?: boolean | string;
            icon?: string;
        }>>> & import("vue").VNodeProps & import("vue").AllowedComponentProps & import("vue").ComponentCustomProps, never>;
        $attrs: {
            [x: string]: unknown;
        };
        $refs: {
            [x: string]: unknown;
        };
        $slots: Readonly<{
            [name: string]: import("vue").Slot;
        }>;
        $root: import("vue").ComponentPublicInstance<{}, {}, {}, {}, {}, {}, {}, {}, false, import("vue").ComponentOptionsBase<any, any, any, any, any, any, any, any, any, {}>>;
        $parent: import("vue").ComponentPublicInstance<{}, {}, {}, {}, {}, {}, {}, {}, false, import("vue").ComponentOptionsBase<any, any, any, any, any, any, any, any, any, {}>>;
        $emit: (event: string, ...args: any[]) => void;
        $el: any;
        $options: import("vue").ComponentOptionsBase<Readonly<import("vue").ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{
            onClick?: () => void;
            value?: string;
            enabled?: boolean;
            loading?: boolean | string;
            icon?: string;
        }>>>, {}, unknown, {}, {}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, {}, string, {}> & {
            beforeCreate?: (() => void) | (() => void)[];
            created?: (() => void) | (() => void)[];
            beforeMount?: (() => void) | (() => void)[];
            mounted?: (() => void) | (() => void)[];
            beforeUpdate?: (() => void) | (() => void)[];
            updated?: (() => void) | (() => void)[];
            activated?: (() => void) | (() => void)[];
            deactivated?: (() => void) | (() => void)[];
            beforeDestroy?: (() => void) | (() => void)[];
            beforeUnmount?: (() => void) | (() => void)[];
            destroyed?: (() => void) | (() => void)[];
            unmounted?: (() => void) | (() => void)[];
            renderTracked?: ((e: import("vue").DebuggerEvent) => void) | ((e: import("vue").DebuggerEvent) => void)[];
            renderTriggered?: ((e: import("vue").DebuggerEvent) => void) | ((e: import("vue").DebuggerEvent) => void)[];
            errorCaptured?: ((err: unknown, instance: import("vue").ComponentPublicInstance<{}, {}, {}, {}, {}, {}, {}, {}, false, import("vue").ComponentOptionsBase<any, any, any, any, any, any, any, any, any, {}>>, info: string) => boolean | void) | ((err: unknown, instance: import("vue").ComponentPublicInstance<{}, {}, {}, {}, {}, {}, {}, {}, false, import("vue").ComponentOptionsBase<any, any, any, any, any, any, any, any, any, {}>>, info: string) => boolean | void)[];
        };
        $forceUpdate: () => void;
        $nextTick: typeof import("vue").nextTick;
        $watch(source: string | Function, cb: Function, options?: import("vue").WatchOptions<boolean>): import("vue").WatchStopHandle;
    } & Readonly<import("vue").ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{
        onClick?: () => void;
        value?: string;
        enabled?: boolean;
        loading?: boolean | string;
        icon?: string;
    }>>> & import("vue").ShallowUnwrapRef<{}> & {} & import("vue").ComponentCustomProperties;
    __isFragment?: never;
    __isTeleport?: never;
    __isSuspense?: never;
} & import("vue").ComponentOptionsBase<Readonly<import("vue").ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{
    onClick?: () => void;
    value?: string;
    enabled?: boolean;
    loading?: boolean | string;
    icon?: string;
}>>>, {}, unknown, {}, {}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, {}, string, {}> & import("vue").VNodeProps & import("vue").AllowedComponentProps & import("vue").ComponentCustomProps & (new () => {
    $slots: {
        default: (_: {}) => any;
    };
});
export default _default;
declare type __VLS_NonUndefinedable<T> = T extends undefined ? never : T;
declare type __VLS_TypePropsToRuntimeProps<T> = {
    [K in keyof T]-?: {} extends Pick<T, K> ? {
        type: import('vue').PropType<__VLS_NonUndefinedable<T[K]>>;
    } : {
        type: import('vue').PropType<T[K]>;
        required: true;
    };
};
