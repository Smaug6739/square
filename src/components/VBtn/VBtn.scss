button {
	display: inline-block;
	font-weight: 400;
	text-align: center;
	vertical-align: middle;
	border: 1px solid transparent;
	padding: 0.375rem 0.75rem;
	margin: 0 0.300rem 0 0;
	font-size: 1rem;
	line-height: 1.5;
	border-radius: 0.25rem;
	transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out, box-shadow .15s ease-in-out;
	cursor: pointer;
}

button:hover {
	opacity: 0.8;
}

button[disabled] {
	opacity: 0.6;
	cursor: not-allowed !important;
}