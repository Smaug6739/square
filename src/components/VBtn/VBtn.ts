import { defineProps, defineComponent } from 'vue';

defineProps<{
  onClick: () => void;
  value?: string;
  enabled?: boolean;
  loading?: boolean | string;
  icon: string;
}>();

