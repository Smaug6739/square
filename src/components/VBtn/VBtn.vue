<template>
	<p>
		Test 2
	</p>
</template>

<script>
import { defineComponent } from 'vue';
export default defineComponent({
	name: 'VBtn',
});
</script>