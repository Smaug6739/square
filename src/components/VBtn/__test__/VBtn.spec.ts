import { describe, expect, test } from 'vitest';
import { mount } from '@vue/test-utils';
import VBtn from '../VBtn.vue';

describe('VBtn snapshot', () => {
  test('default', () => {
    const wrapper = mount(VBtn);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
