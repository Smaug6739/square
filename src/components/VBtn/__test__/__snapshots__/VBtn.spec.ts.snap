// Vitest Snapshot v1

exports[`VBtn snapshot > default 1`] = `"<p> Test 2 </p>"`;
